### PR TITLE
Fix target can't be found if target is app

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
Fix https://github.com/rancher/rio/issues/906.

File: cli/cmd/publicdomain/add.go [hunk 1]
```diff
@@ -22,30 +22,41 @@ func (r *Register) Run(ctx *clicontext.CLIContext) error {
...
-		svc := result.Object.(*riov1.Service)
-		targetApp, targetVersion := services.AppAndVersion(svc)
-		if !strings.Contains(target, "@") {
-			targetVersion = ""
+
+	/*
+		1) if target is [namespace:]name and can't be found as an object, treat it as app
+		2) if target is [namespace:]name@version, treat it as service(revision)
+		3) if target is [namespace:]name and can be found as an object, treat it as a router
+	*/
```
Comment (cbron) [inline cli/cmd/publicdomain/add.go]: Nice comments. Would be nice to add unit or integration tests to this in long term.

Comment (StrongMonkey) [reply cli/cmd/publicdomain/add.go]: We can cover that in validation tests. I think all public domain tests are validation tests.

Comment (cbron) [reply cli/cmd/publicdomain/add.go]: cc @rancher-max ^ low priority, similar to rio build cases.

**Pull request merged**